File: n226-invert-binary-tree.go
package main

func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	left := invertTree(root.Right)
	right := invertTree(root.Left)

	root.Left = left
	root.Right = right

	return root
}

func invertTree2(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	tree := TreeNode{
		Val:   root.Val,
		Left:  invertTree(root.Right),
		Right: invertTree(root.Left),
	}

	return &tree
}

File: n226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    if (root !== null) {
        var left = invertTree(root.right)
        var right = invertTree(root.left)
        
        root.left = left
        root.right = right
    }
    
    return root
};
